### ISSUE TITLE
crash when using MythTV WebFrontend

### ISSUE THREAD

**Issue by tdufaure** [MythTV/mythtv#306]
* **Platform**:
```
root@:~# lsb_release -a
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.1 LTS
Release:        20.04
Codename:       focal
root@:~# uname -a
Linux 5.4.0-60-generic #67-Ubuntu SMP Tue Jan 5 18:31:36 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
```

* **MythTV version**: 31

* **Package version**: `2:31.0+fixes.20200323.9579662cdc-0ubuntu1`

* **Component**: `mythtv-backend` & `mythtv-frontend`

### What steps will reproduce the bug?

Whenever I use the `MythTV WebFrontend` to edit a recording rule (via `TV` \ `Recording Rules` \ `Edit Recording Rules`), whatever change I make, when pressing `Save Changes`, the `MythTV WebFrontend` returns the message `Save Failed!` and the backend seems to crash and restart.

In `/var/log/mythtv/mythbackend.log`, I found those messages :
```
Jan 25 21:51:24 teaghome mythbackend: mythbackend[173206]: C CoreContext signalhandling.cpp:299 (handleSignal) Received Erreur de segmentation: Code 1, PID 53, UID 0, Value 0x00000000
Jan 25 21:51:26 teaghome mythbackend: mythbackend[175179]: C thread_unknown mythcommandlineparser.cpp:2602 (ConfigureLogging) mythbackend version: fixes/31 [v31.0-v31.0] www.mythtv.org
Jan 25 21:51:26 teaghome mythbackend: mythbackend[175179]: C thread_unknown mythcommandlineparser.cpp:2606 (ConfigureLogging) Qt version: compile: 5.12.5, runtime: 5.12.8
Jan 25 21:51:26 teaghome mythbackend: mythbackend[175179]: I thread_unknown mythcommandlineparser.cpp:2608 (ConfigureLogging) Ubuntu 20.04.1 LTS (x86_64)
Jan 25 21:51:26 teaghome mythbackend: mythbackend[175179]: N thread_unknown mythcommandlineparser.cpp:2610 (ConfigureLogging) Enabled verbose msgs:  general
```

### How often does it reproduce? Is there a required condition?

Every time I use the `MythTV WebFrontend` to edit a recording rule.

### What is the expected behaviour?

Saving my changes ! :)

### What do you see instead?

Message `Save Failed!` and a restart of the backend.

**Comment by tdufaure** [comment]
Tried a :
`/usr/bin/mythbackend --loglevel debug`

Not really much information : 
```2021-01-28 21:11:37.765493 D  Running HouseKeeper.
2021-01-28 21:11:37.765509 D  Checking to run DBCleanup
2021-01-28 21:11:37.765520 D  Checking to run HardwareProfiler
2021-01-28 21:11:37.765548 D  Checking to run JobQueueRecover
2021-01-28 21:11:37.765556 D  Checking to run LogClean
2021-01-28 21:11:37.765563 D  Checking to run MythFillDB
2021-01-28 21:11:37.765617 D  MythFillDatabase scheduled to run at ven. janv. 29 11:10:15 2021 GMT.
2021-01-28 21:11:37.765627 D  Checking to run RecordedArtworkUpdate
2021-01-28 21:11:37.765641 D  Checking to run ThemeUpdateNotifications
2021-01-28 21:11:41.211802 C  Received Erreur de segmentation: Code 1, PID 53, UID 0, Value 0x00000000```

Too bad ... :-/

**Comment by billmeek** [comment]
I'd suggest restarting the backend after adding the following to
/etc/mythtv/additional.args:
```
ADDITIONAL_ARGS="-v http:debug,upnp:debug"
```
I didn't drop core when trying this on master.

Simply comment out the ADDITIONAL_ARGS line so you won't have the additional
logging next time systemd starts the BE. **mythbackend --serverbose http:alert,upnp:alert**
should turn it off **temporarily** without restarting.

**Comment by stuarta** [comment]
We are currently working on improving the whole of this part of MythTV for the next version

**Comment by gloomytrousers** [comment]
I see this on Fedora 35 with the MythTV package from RPMFusion: mythtv-backend-31.0-20.158.20210711git0680b37c68.fc35.x86_64

The journal contains the line:
```
Dec 05 14:58:55 host.example.com mythbackend[539036]: 2021-12-05 14:58:55.377005 C [539036/539036] CoreContext signalhandling.cpp:299:handleSignal  Received Segmentation fault: Code 1, PID 69, UID 0, Value 0x00000000
```

**Comment by garybuhrmaster** [comment]
> I see this on Fedora 35 with the MythTV package from RPMFusion: mythtv-backend-31.0-20.158.20210711git0680b37c68.fc35.x86_64

First, the RPMFusion version of MythTV is rather dated.  One should request the packager to update to a recent fixes release or build/install your own version (I do not know if any fixes address the issue, but reproducers need to be on the latest to avoid wasting developers time).

Secondly, after the update, you should add in the debug options as specified by billmeek to provide additional appropriate debugging, and provide the full backtrace (instructions are on the wiki how to provide useful information to the developers).

**Comment by gloomytrousers** [comment]
Thanks, [requested](https://bugzilla.rpmfusion.org/show_bug.cgi?id=6164). Will update here, with detailed logs, if I can still replicate on a newer version.

**Comment by paul-h** [comment]
There is also a thread on the forum that also has a backtrace
https://forum.mythtv.org/viewtopic.php?f=36&t=4667#p22981

**Comment by rcrdnalor** [comment]
@tdufaure : WebFrontend was replaced by Web Application (https://www.mythtv.org/wiki/Web_Application)
in MythTv v34 and MythTV master (v35-pre).
Is this issue still valid on recent installations of MythTv ?

**Comment by gloomytrousers** [comment]
@rcrdnalor: No, I have had no crashes in the new version, and I use it a fair amount. I got them often and predictably on the old one.

**Comment by rcrdnalor** [comment]
Closing this issue upon feedback from originator (see https://github.com/MythTV/mythtv/issues/306#issuecomment-2603168529)
